resources/knative-eventing/charts/knative-eventing/templates/pre-upgrade-job.yaml

```diff
@@ -36,29 +45,109 @@ data:
   pre-upgrade.sh: |
     #!/usr/bin/env bash
     set -eu
+    RED='\033[0;31m'
+    GREEN='\033[0;32m'
+    NC='\033[0m' # No Color
+
+    function shout() {
+        echo -e "${GREEN}
+    #################################################################################################
+    # $1
+    #################################################################################################
+        ${NC}"
+    }
+
+    function shoutFail() {
+        echo -e "${RED}
+    #################################################################################################
+    # $1
+    #################################################################################################
+        ${NC}"
+    }
+
+    function removeFinalizersFromSubscriptions() {
+      # Removal of finalizers is needed to delete Subscriptions
+      shout "+ Patching all subscriptions"
+      ignoredNamespaces='kyma-system|istio-system|knative-eventing|knative-serving|kube-node-lease|kube-public|kube-system|kyma-installer|natss'
+      for ns in $(kubectl get ns --no-headers | grep -vE "${ignoredNamespaces}" | awk '{print $1}'); do
+         for sub in $(kubectl get -n "${ns}" subscriptions.messaging.knative.dev --ignore-not-found -ojsonpath='{.items[*].metadata.name}'); do
+           if [[ ! -z "${sub}" ]]; then
+             shout "++ Patching subscription: ${sub} in ns: ${ns}"
+             kubectl patch -n "${ns}" subscriptions.messaging.knative.dev "${sub}" --type merge -p '{"metadata": {"finalizers": []}}'
```

Comment:
set -e will fail this script hard at that point in case of an error . is that really what we want?

Reply:
Yes, if there are subscriptions, they need to be patched. If there's an error the job should fail fast.